src/main/webapp/js/query.js

```diff
@@ -325,6 +325,157 @@ function createSearchElement(search) {
+            if (submitButtonEl != null) {
+                submitButtonEl.remove();
+            }
+            modal.style.display = "none";
+        }
+        // When the user clicks anywhere outside of the modal, close it
+        window.onclick = function(event) {
+            if (event.target == modal) {
+                let restaurantContainerEl = document.getElementById("restaurant-name-container");
+                restaurantContainerEl.remove();
+                let submitButtonEl = document.getElementById("submit-button");
+                if (submitButtonEl != null) {
+                    submitButtonEl.remove();
+                }
+                modal.style.display = "none";
+            }
+        }
+        feedbackButton = document.createElement('button');
+        feedbackButton.className = 'btn1 feedback';
+        feedbackButton.innerText = "Submit Feedback";
+        feedbackButton.addEventListener('click', () => {
+            let restaurantNameEl = createRestaurantElement(search.name);
+            formEl.appendChild(restaurantNameEl);
+            modal.style.display = "block";
+        });
+        newCardBody.appendChild(feedbackButton);
+    }
+    let searchButton = document.createElement('button');
+    searchButton.className = 'btn1 search';
+    searchButton.innerText = "Search with These Parameters Again";
+    newCardBody.appendChild(searchButton);
+    searchButton.addEventListener('click', () => {
+        reroll()});
+    return newCardBody;
+}
+
+//Function to append restaurant name to modal form to force it to follow through to feedback
+function createRestaurantElement(restaurantName) {
+    let userID = 0;
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    let userEl = document.createElement('input');
+    userEl.className = "input--style-2";
+    userEl.type = "text";
+    userEl.id = "user-id";
+    userEl.name = "user-id";
+    userEl.value = userID;
+    userEl.hidden = true;
+
+    let inputGroupEl = document.createElement('div');
+    inputGroupEl.className = "input-group";
+    inputGroupEl.id = "restaurant-name-container";
+    let inputContainer = document.createElement('input');
+    inputContainer.className = "input--style-2";
+    inputContainer.type = "text";
+    inputContainer.id = "restaurant-name-fill";
+    inputContainer.name = "restaurant-name-fill";
+    inputContainer.value = restaurantName;
+    inputContainer.innerText = restaurantName;
+    inputGroupEl.appendChild(inputContainer);
+    inputGroupEl.appendChild(userEl);
+
+    let submitEl = document.createElement('input');
+    submitEl.type = "submit";
+    submitEl.id = "submit-button";
+    inputGroupEl.appendChild(submitEl);
+    return inputGroupEl;
+}
+
+async function fetchFeedback() {
+    let userID = 0;
+    if (localStorage.getItem("loggedIn")){
+        userID = localStorage.getItem("user");
+    }
+    let response = await fetch(`/feedback?user=${userID}`, {
```

Comment:
do you just mean to chain these all together? this await pauses the execution of the function. Once the fetch finishes, then we attach the `then`s. Instead, we can do just

return fetch(...).then(...).then(...);

Reply:
Chain all of the async functions together? I was having trouble getting the results back in the createSearchElement function. I got errors because the .thens took a little too long to finish and so when I called them in createSearchElement I got an error (something along the lines of waiting for the promise, and then it couldn't complete so other things would be assigned to null).

Reply:
in this case,

```
let response = await ...;
return response
```

is equivalent to

```
return await fetch(...).then(...)...;
```

and also in this case, since we don't do special handling on rejection, it's also equivalent to writing a return statement without `await`:

```
return fetch(...).then(...)...;
```

Check out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Rewriting_a_Promise_chain_with_an_async_function.